### ISSUE TITLE
Attempted to acces a missing method

### ISSUE THREAD

**Issue by arjen-hiemstra** [firegiant/HeatWaveSupport#59]
### HeatWave Version

1.0.3

### Visual Studio Version

17.6.5

### Repro Steps

1. Install HeatWave
2. Try to create a new MSI package project.

### Actual Result

I get an error from VS about missing a method.
![image](https://github.com/firegiant/HeatWaveSupport/assets/59915285/974f4378-8cf8-4d4d-bdfd-14ba9b39f058)

From the log I can see this: 
```log
=====================
02/04/2024 10:33:30
NotRecoverable
Microsoft.VisualStudio.Composition.CompositionFailedException: An exception was thrown while initializing part "FireGiant.HeatWave.ProjectSystem.HeatWaveProjectTreeIconPropertyProvider".
   at Microsoft.VisualStudio.Composition.RuntimeExportProviderFactory.RuntimeExportProvider.RuntimePartLifecycleTracker.CreateValue()
   at Microsoft.VisualStudio.Composition.ExportProvider.PartLifecycleTracker.Create()
   at Microsoft.VisualStudio.Composition.ExportProvider.PartLifecycleTracker.MoveNext(PartLifecycleState nextState)
   at Microsoft.VisualStudio.Composition.ExportProvider.PartLifecycleTracker.MoveToState(PartLifecycleState requiredState)
   at Microsoft.VisualStudio.Composition.ExportProvider.PartLifecycleTracker.GetValueReadyToExpose()
   at Microsoft.VisualStudio.Composition.RuntimeExportProviderFactory.RuntimeExportProvider.ConstructExportedValue(RuntimeImport import, RuntimeExport export, RuntimePartLifecycleTracker importingPartTracker, PartLifecycleTracker partLifecycle, ReportFaultCallback faultCallback)
   at Microsoft.VisualStudio.Composition.RuntimeExportProviderFactory.RuntimeExportProvider.<>c__DisplayClass16_0.<GetExportedValue>b__1()
   at Microsoft.VisualStudio.Composition.DelegateServices.<>c__DisplayClass2_0`1.<As>b__0()
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at Microsoft.VisualStudio.ProjectSystem.MefExtensions.<ExtensionValuesWithMetadata>d__12`2.MoveNext()
   at Microsoft.VisualStudio.ProjectSystem.DualOrderPrecedenceImporter`2.Merge(IReadOnlyCollection`1 source1, IReadOnlyCollection`1 source2)
   at Microsoft.VisualStudio.ProjectSystem.DualOrderPrecedenceImporter`2.<LinkExternalInput>b__9_0(IProjectVersionedValue`1 data)
   at Microsoft.VisualStudio.ProjectSystem.TransformBlockSlim`2.TransformBlockSlimSync.ProcessInputAsync(TInput input)
   at Microsoft.VisualStudio.ProjectSystem.DataReceivingBlockSlim`1.<ProcessInputQueueAsync>d__5.MoveNext()
---> System.MissingMethodException: Method not found: 'Microsoft.VisualStudio.Threading.JoinableTaskFactory Microsoft.VisualStudio.ProjectSystem.IProjectThreadingService.get_JoinableTaskFactory()'.
   at FireGiant.HeatWave.ProjectSystem.HeatWaveProjectTreeIconPropertyProvider..ctor(UnconfiguredProject unconfiguredProject, IProjectThreadingService threadingService)
--- End of inner exception stack trace ---
CPS Microsoft.VisualStudio.ProjectSystem.RetailRuntime RunningInVisualStudio HostSetActiveProjectConfiguration AllTargetOutputGroups DeclaredSourceItems VisualStudioWellKnownOutputGroups SdkReferences HeatWaveProject ProjectReferences UserSourceItems SortByDisplayOrder UseFileGlobs OpenProjectFile PackageReferences ProjectConfigurationsDeclaredDimensions ProjectImportsTree OutputGroups DependenciesTree HandlesOwnReload .NET FallbackProjectConfigurationService 
===================
```

### Expected Result

I was expecting to be able to create a new project, or load an existing one. Both situations lead to this error.

**Comment by FireGiantHelp** [comment]
Please try upgrading your Visual Studio 2022 to 17.9.

We've opened an issue for our dev team to investigate the possibility of blocking the installation of HeatWave on old versions of Visual Studio 2022.

**Comment by arjen-hiemstra** [comment]
That solved it.

**Comment by dcm99** [comment]
@FireGiantHelp thanks for the workaround but discontinuing support for VisualStudio 17.8 seems a bit harsh. 17.8 is an LTS version and 17.9 is not. I know a lot of developers who skip the non-LTS versions, and in my company, the IT department only allows the non-LTS versions in isolated environments.
Edit:
I just saw that this Issue is about VS 17.6 but I can confirm that the Problem is also present in 17.8.8

**Comment by FireGiantHelp** [comment]
We would like to support the latest Visual Studio LTS. Unfortunately, due to a breaking change in the Visual Studio SDK, we have a dependency on 17.9. We are working with the Visual Studio team to see what is possible.

**Comment by FireGiantHelp** [comment]
FYI: We just released HeatWave v1.0.4, which supports the current Visual Studio version *and* Visual Studio's latest LTS. That means HeatWave should work with the latest VS 17.8 as well as the latest VS 17.9.

**Comment by D3vil0p3r** [comment]
I have the same issue on VS 17.11.
